### ISSUE TITLE
Charakterystyka przewyższeń

### ISSUE THREAD

**Issue by zaquncio** [edk-software/rejony-app#44]
Zapis obliczonej przez weryfikator charakterystyki przewyższeń.

**Comment by zaquncio** [comment]
@duzymaju dasz radę?

**Comment by duzymaju** [comment]
Mogę prosić o dokładniejszy opis? Jaki ma być adres akcji w API, jakiego JSONa ma przyjmować, co walidować, gdzie i w jakiej postaci zapisywać.

**Comment by zaquncio** [comment]
Do tabeli tras trzeba dodać jedną kolumnę typu pewnie text w której będzie można zapisać JSON'a. W controllerze tras metoda API, która przyjmie obiekt JSON i zapisze go do bazy.

Automatyczna weryfikacja śladu wylicza histogram przewyższeń, chodzi o to, aby weryfikator mógł zapisać to co sobie przygotuje do wizualizacji w bazie. Potem strona będzie z tego korzystać to narysowania takiego samego histogramu. Głownie chodzi o to aby nie wyliczać go kolejny raz.

Metoda w weryfikatorze: Patrz: metoda [drawElevationChart w Controls.js.](https://github.com/qooban/edk-route-verifier/blob/master/src/Controls.js)

**Comment by duzymaju** [comment]
@qooban: Przygotowuję akcję do dodawania charakterystyki (JSON) do tabeli tras. Musimy ustalić metodę jej zabezpieczenia przed nieuprawnionym użyciem. Nie musimy szyfrować przesyłanych danych (rozumiem, że nie będzie tam niczego "wrażliwego") ale na pewno potrzebujemy klucza, którym zautoryzujemy próbę update'u rekordu trasy - może być generowany dynamicznie, może być stały, zaszyty w zmiennych środowiskowych - co myślisz?

**Comment by qooban** [comment]
@duzymaju: Jak dla mnie może być stały. A jak jest autoryzowane zatwierdzanie trasy (edk_route_approve)? Jest tam jakiś mechanizm zabezpieczający?

**Comment by duzymaju** [comment]
Dodałem zwykłe BasicAuth: #80.

**Comment by zaquncio** [comment]
@ jakie uprawnienia ma mieć ten user?

**Comment by qooban** [comment]
@duzymaju Tak, `data` zawiera główne dane. Myślę, że warto będzie przechowywać `data` i  `labels`.

**Comment by duzymaju** [comment]
@zaquncio: Jego dodajesz tak, jak napisałem w opisie #80, tylko w `parameters.yml` - login, hasło i rolę `ROLE_ROUTE_EDITOR`.

**Comment by zaquncio** [comment]
Rozumiem ze musi to byc uzytkownik zarejestrowany w systemie? To jest nowa
rola na projekcie? Gdzie sie ja nadaje?

03.03.2018 10:17 AM "Wiktor Maj" <notifications@github.com> napisał(a):

**Comment by duzymaju** [comment]
Nie - jedyne, co musisz zrobić, to do pliku `parameters.yml` na każdym środowisku dodać:
```
    basic_auth_users:
        route.verifier:
            password: password
            roles: ROLE_ROUTE_EDITOR
```
jeśli użytkownik ma mieć login `route.verifier` i zmienić `password` na jakiś ciąg znaków. I tyle, bo w `security.yml` dodałem providera `memory_provider`, który importuje sobie tę listę użytkowników (tu z jednym, jedynym użytkownikiem `route.verifier`), a później ten provider jest wykorzystywany w firewallu, który chroni naszą akcję, więc użytkownik `route.verifier`, który posiada rolę `ROLE_ROUTE_EDITOR` wymaganą przez kontrolę dostępu do tej akcji, będzie miał do niej dostęp.

Tak więc dodaj tylko dane użytkownika do `parameters.yml` i tyle. :)

**Comment by duzymaju** [comment]
Dobra, teraz uświadomiłem sobie, co zrobiłem... Pisałem tę zmianę zanim dowiedziałem się, że `edk-route-verifier` nie jest osobną aplikacją, tylko narzędziem dewelopowania jednego ze skryptów JS dla `rejony-app`, więc założyłem, że potrzebujemy użytkownika do komunikacji backendu jednej aplikacji z drugą... :P Przepraszam - mój błąd. Zaoram zaraz tę kontrolę użytkowników i dodam normalne uprawnienia dla użytkowników z bazy i nie będzie żadnego BasicAuth.

**Comment by duzymaju** [comment]
OK, gotowe - teraz wystarczy request zatwierdzania trasy wysłać metodą `POST` zamiast `GET` i w treści przekazać `JSON` z charakterystyką przewyższeń a zostanie on zapisany w bazie. Nadal można oczywiście pchać też `GET`y ale w ich przypadku pole charakterystyki przewyższeń pozostanie puste. BTW: zgodnie ze sztuką powinniśmy w `rejony-app` zdefiniować metodę, którą będzie się zatwierdzało trasę i nie powinien być to `GET` (służący do pobrania danych), tylko `POST`, a prędzej nawet `PATCH` (bo dane zostają zmodyfikowane). Co myślisz @zaquncio?

**Comment by zaquncio** [comment]
ok.

W dniu 4 marca 2018 01:40 użytkownik Wiktor Maj <notifications@github.com>
napisał:

**Comment by zaquncio** [comment]
Jasne
